awkward's named axes causes regression in test_notebooks

### Vector Version

1.5.2

### Python Version

3.11 and 3.12

### OS / Environment

nixpkgs
https://github.com/NixOS/nixpkgs/pull/354346

### Describe the bug

Failure in `tests/test_notebooks.py`
```output
E           papermill.exceptions.PapermillExecutionError: 
E           ---------------------------------------------------------------------------
E           Exception encountered at "In [96]":
E           ---------------------------------------------------------------------------
E           AttributeError                            Traceback (most recent call last)
E           Cell In[96], line 2
E                 1 (
E           ----> 2     vector.Array(
E                 3         [  # an Awkward Array of vectors
E                 4             [{"x": 1, "y": 1.1}, {"x": 2, "y": 2.2}],
E                 5             [],
E                 6             [{"x": 3, "y": 3.3}],
E                 7             [{"x": 4, "y": 4.4}, {"x": 5, "y": 5.5}],
E                 8         ]
E                 9     )
E                10     + vector.obj(x=10, y=5)  # and a single vector object
E                11 )
E           
E           File /nix/store/yq4a8j78af4l3rcc1m7sf5kqqv3ydnkb-python3.12-awkward-2.7.0/lib/python3.12/site-packages/awkward/_operators.py:54, in _binary_method.<locals>.func(self, other)
E                51 if _disables_array_ufunc(other):
E                52     return NotImplemented
E           ---> 54 return ufunc(self, other)
E           
E           File /nix/store/yq4a8j78af4l3rcc1m7sf5kqqv3ydnkb-python3.12-awkward-2.7.0/lib/python3.12/site-packages/awkward/highlevel.py:1594, in Array.__array_ufunc__(self, ufunc, method, *inputs, **kwargs)
E              1592 name = f"{type(ufunc).__module__}.{ufunc.__name__}.{method!s}"
E              1593 with ak._errors.OperationErrorContext(name, inputs, kwargs):
E           -> 1594     return ak._connect.numpy.array_ufunc(ufunc, method, inputs, kwargs)
E           
E           File /nix/store/yq4a8j78af4l3rcc1m7sf5kqqv3ydnkb-python3.12-awkward-2.7.0/lib/python3.12/site-packages/awkward/_connect/numpy.py:367, in array_ufunc(ufunc, method, inputs, kwargs)
E               364 attrs = attrs_of(*inputs)
E               365 backend = backend_of(*inputs, coerce_to_common=True)
E           --> 367 depth_context, lateral_context = NamedAxesWithDims.prepare_contexts(inputs)
E               369 inputs = _array_ufunc_custom_cast(inputs, behavior, backend)
E               371 def action(inputs, **ignore):
E           
E           File /nix/store/yq4a8j78af4l3rcc1m7sf5kqqv3ydnkb-python3.12-awkward-2.7.0/lib/python3.12/site-packages/awkward/_namedaxis.py:686, in NamedAxesWithDims.prepare_contexts(cls, arrays, unwrap_kwargs)
E               684         layout = ctx.unwrap(array, **_unwrap_kwargs)
E               685     _named_axes.append(_get_named_axis(array))
E           --> 686     _ndims.append(layout.minmax_depth[1])
E               688 depth_context = {NAMED_AXIS_KEY: cls(_named_axes, _ndims)}
E               689 lateral_context = {NAMED_AXIS_KEY: cls(_named_axes, _ndims)}
E           
E           AttributeError: 'VectorObject2D' object has no attribute 'minmax_depth'
E           
E           This error occurred while calling
E           
E               numpy.add.__call__(
E                   <VectorArray2D [[{x: 1, y: 1.1}, {...}], ...] type='4 * var * Vecto...'>
E                   VectorObject2D-instance
E               )
```

### Any additional but relevant log output

_No response_

--- a/src/awkward/_namedaxis.py
+++ b/src/awkward/_namedaxis.py
@@ -652,7 +652,7 @@ class NamedAxesWithDims:
     """
 
     named_axis: list[AxisMapping]
-    ndims: list[int]
+    ndims: list[int | None]
 
     def __post_init__(self):
         if len(self.named_axis) != len(self.ndims):
@@ -660,7 +660,7 @@ class NamedAxesWithDims:
                 "The number of dimensions must match the number of named axis mappings."
             )
 
-    def __iter__(self) -> tp.Iterator[tuple[AxisMapping, int]]:
+    def __iter__(self) -> tp.Iterator[tuple[AxisMapping, int | None]]:
         yield from zip(self.named_axis, self.ndims)
 
     @classmethod
@@ -683,18 +683,20 @@ class NamedAxesWithDims:
             with HighLevelContext() as ctx:
                 layout = ctx.unwrap(array, **_unwrap_kwargs)
             _named_axes.append(_get_named_axis(array))
-            _ndims.append(layout.minmax_depth[1])
+            _ndims.append(getattr(layout, "minmax_depth", (None, None))[1])
 
         depth_context = {NAMED_AXIS_KEY: cls(_named_axes, _ndims)}
         lateral_context = {NAMED_AXIS_KEY: cls(_named_axes, _ndims)}
         return depth_context, lateral_context
 
-    def __setitem__(self, index: int, named_axis_with_ndim: tuple[AxisMapping, int]):
+    def __setitem__(
+        self, index: int, named_axis_with_ndim: tuple[AxisMapping, int | None]
+    ):
         named_axis, ndim = named_axis_with_ndim
         self.named_axis[index] = named_axis
         self.ndims[index] = ndim
 
-    def __getitem__(self, index: int) -> tuple[AxisMapping, int]:
+    def __getitem__(self, index: int) -> tuple[AxisMapping, int | None]:
         return self.named_axis[index], self.ndims[index]
 
     def __len__(self) -> int:

